products/views.py

```diff
@@ -1,3 +1,276 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.utils  import IntegrityError
+from django.db.models import Q
+
+from postings.models import (
+    Comment,
+    Posting
+)
+
+from products.models import (
+    Category,
+    Color,
+    DetailedProduct,
+    Image,
+    Menu,
+    Product,
+    Size
+)
+
+from users.models    import User
+
+class MenuView(View) :
+    def post(self, request) :
+        try :
+            data      = json.loads(request.body)
+            menu_name = data['name']
+
+            Menu.objects.create(name=menu_name)
+
+            return JsonResponse({'message':'Save Success'}, status=201)
+        
+        except KeyError as e : 
+            return JsonResponse({'message': e}, status=400)
+
+    def get(self, request) :
+        try :
+            menus      = Menu.objects.all()
+            menu_lists = []
+
+            for menu in menus :
+
+                menu_lists.append({
+                    'id'   : menu.id ,
+                    'name' : menu.name
+                })
```

Comment:
- 항상 코드 push 하시기전에 이런 불필요한 공백, 컴마 사이 공백 여부 등등 확인하시고 한사람이 작성한 것 처럼 통일되어 있는지 확인해주세요.
- 해당 for loop 은 list comprehension 으로 대체 가능합니다. list comprehension 으로 바꿔보시고, 기존 for loop 에서 list comprehension 으로 변경했을때 어떤 이점이 있는지 댓글 달아주세요.
```suggestion
            for menu in menus :
                menu_lists.append({
                    'id'   : menu.id,
                    'name' : menu.name
                })
```

Reply:
사용해보니 기존의 list에 값을 추가하는거보다 간결하고 편리합니다.

또한 불필요한 메서드의 사용을 줄일 수 있었습니다.